### DISCUSSION TITLE
how to limit the number of freerdp connection ？

### DISCUSSION THREAD

**Discussion by 407219125** [FreeRDP/FreeRDP#9055] [General]
I use freerdp to connect from Windows to Ubuntu, and I can start many connections. How can I limit the number of connections?

**Comment by hardening** [comment]
Can you elaborate ? Where do you want to impose the limit (client or server) ? Is it the global number of connections or connections per user ?

**Reply by 407219125** [reply]
global number of connections

**Comment by akallabeth** [comment]
> I use freerdp to connect from Windows to Ubuntu, and I can start many connections. How can I limit the number of connections?

so, you are using `wfreerdp.exe` client on windows or you use `ogon`, `gnome-remote-desktop` or `freerdp-shadow` on your linux machine as RDP server?

**Reply by 407219125** [reply]
i use wfreerdp.exe client on windows, freerdp-shadow-cli on linux.

**Reply by akallabeth** [reply]
`freerdp-shadow-cli` does currently not have a limitation for connected clients, but that might be something that could be added.

**Reply by 407219125** [reply]
ok, thank you.

**Reply by akallabeth** [reply]
@407219125 I did like the idea, did implement a simple connection limit with https://github.com/FreeRDP/FreeRDP/pull/9056

**Reply by 407219125** [reply]
I really appreciate it!
